legal-api/src/legal_api/services/compliance/compliance_checks/firms.py

```diff
@@ -0,0 +1,254 @@
+
+
+def check_compliance(business: Business) -> list:
+    """Check for non-compliant business data."""
+    result = []
+
+    legal_type = business.legal_type
+
+    result.extend(check_office(business))
+    result.extend(check_parties(legal_type, business))
+
+    return result
+
+
+def check_office(business: Business) -> list:
+    """Check for non-compliant office data."""
+    result = []
+
+    business_office = business.offices \
+        .filter(Office.office_type == 'businessOffice') \
+        .one_or_none()
+
+    if not business_office:
+        result.append({
+            'code': ComplianceWarningCodes.NO_BUSINESS_OFFICE,
+            'message': 'A business office is required.',
+        })
+        return result
+
+    mailing_address = business_office.addresses \
+        .filter(Address.address_type == 'mailing') \
+        .one_or_none()
+    result.extend(check_address(mailing_address, Address.MAILING, ComplianceWarningReferers.BUSINESS_OFFICE))
+
+    delivery_address = business_office.addresses \
+        .filter(Address.address_type == 'delivery') \
+        .one_or_none()
+    result.extend(check_address(delivery_address, Address.DELIVERY, ComplianceWarningReferers.BUSINESS_OFFICE))
+
+    return result
+
+
+def check_parties(legal_type: str, business: Business) -> list:
+    """Check for non-compliant parties data."""
+    result = []
+
+    firm_party_roles = business.party_roles.all()
+    result.extend(check_firm_parties(legal_type, firm_party_roles))
+
+    completing_party_filing = Filing \
+        .get_most_recent_legal_filing(business.id, 'conversion')
+
+    if not completing_party_filing:
+        completing_party_filing = Filing \
+            .get_most_recent_legal_filing(business.id, 'registration')
+
+    result.extend(check_completing_party_for_filing(legal_type, completing_party_filing))
+    return result
+
+
+def check_firm_parties(legal_type: str, party_roles: list) -> list:
+    """Check for non-compliant firm parties data."""
+    result = []
+
+    proprietor_parties = []
+    partner_parties = []
+
+    for party_role in party_roles:
+        if party_role.role == PartyRole.RoleTypes.PROPRIETOR.value:
+            proprietor_parties.append(party_role.party)
+            result.extend(check_firm_party(legal_type, party_role))
+        if party_role.role == PartyRole.RoleTypes.PARTNER.value:
+            partner_parties.append(party_role.party)
+            result.extend(check_firm_party(legal_type, party_role))
+
+    if legal_type == Business.LegalTypes.SOLE_PROP.value and not proprietor_parties:
+        result.append({
+            'code': ComplianceWarningCodes.NO_PROPRIETOR,
+            'message': 'A proprietor is required.',
+        })
+    elif legal_type == Business.LegalTypes.PARTNERSHIP.value and not partner_parties:
+        result.append({
+            'code': ComplianceWarningCodes.NO_PARTNER,
+            'message': 'A partner is required.',
```

Comment:
I think two partners are required for GP. Is that condition checked

Reply:
No, will add that.

Reply:
Added.  Thanks for catching this!